src/plugins/dashboard/public/application/utils/use/use_dashboard_app_state.tsx

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import EventEmitter from 'events';
```

Comment:
same consideration here. maybe could abstract to some common utility or hooks.

Reply:
This file might be harder to abstract to some common utility since it is more specific to the dashboard app. I wrote it in the same structure as the visualize app but i think there will be more dashboard specific logics added here later.